### PR TITLE
Event - Add "Learn how it works"

### PR DESCRIPTION
**Pull request by KvnMrs**
Issue: https://github.com/blockframes/blockframes/issues/8618

### CODE REVIEW

File: libs/event/src/lib/form/shell/shell.component.html [hunk 1]
```diff
@@ -38,6 +38,10 @@ <h2>Details</h2>
         <a mat-tab-link [routerLink]="tab.path" routerLinkActive #rla="routerLinkActive" [active]="rla.isActive">
           {{ tab.label }}</a>
       </ng-container>
+      <button mat-button class="explanation" (click)="explain()">
+        <mat-icon svgIcon="info"></mat-icon>
+        <span>Learn how it works</span>
+      </button>
```
Comment (phpgeekfr) [inline libs/event/src/lib/form/shell/shell.component.html:44]: button should not be inside `nav`.

you can do :
```

    <div fxLayout fxLayoutAlign="space-between center">
      <nav mat-tab-nav-bar >
        <ng-container *ngFor="let tab of (tabs$ |async)">
          <a mat-tab-link [routerLink]="tab.path" routerLinkActive #rla="routerLinkActive" [active]="rla.isActive">
            {{ tab.label }}</a>
        </ng-container>
      </nav>
      <button mat-button (click)="explain()">
        <mat-icon svgIcon="info"></mat-icon>
        <span>Learn how it works</span>
      </button>
    </div>
```

File: libs/event/src/lib/form/shell/shell.component.scss [hunk 2]
```diff
@@ -35,4 +35,8 @@ form {
 
 mat-spinner {
   margin-top: 48px;
+}
+
+.explanation{
+  margin-inline-start: auto;
```
Comment (phpgeekfr) [inline libs/event/src/lib/form/shell/shell.component.scss:41]: no need cf comment above

File: libs/event/src/lib/form/shell/shell.component.ts [hunk 3]
```diff
@@ -184,4 +185,9 @@ export class EventFormShellComponent implements OnInit, OnDestroy {
     this.errorChipMessage = !videoId ? 'Video file missing' : '';
     this.cdr.markForCheck();
   }
+
+  public explain() {
+    this.dialog.open(ExplanationComponent, { data: createModalData({ type: this.form.value.type }), autoFocus: false, });
```
Comment (phpgeekfr) [inline libs/event/src/lib/form/shell/shell.component.ts:190]:
```suggestion
    this.dialog.open(ExplanationComponent, { data: createModalData({ type: this.form.value.type }), autoFocus: false });
```

File: apps/catalog/catalog/src/app/marketplace/title/avails/explanation/explanation.component.scss [hunk 4]
```diff
@@ -1,3 +1,7 @@
+section{
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/marketplace/title/avails/explanation/explanation.component.scss:1]:
```suggestion
section {
```

File: apps/catalog/catalog/src/app/marketplace/title/avails/explanation/explanation.component.scss [hunk 5]
```diff
@@ -1,3 +1,7 @@
...
+  margin-right: 32px;
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/marketplace/title/avails/explanation/explanation.component.scss:2]: with `margin-right` you don't have the same right spacing as with fxLayout:
image


I think this should be better, but I let you test this and update other component also
```
  display: flex;
  flex-direction: column;
```

File: libs/event/src/lib/explanation/explanation.component.scss [hunk 6]
```diff
@@ -0,0 +1,24 @@
+section{
+  margin-right: 32px;
+}
```
Comment (phpgeekfr) [inline libs/event/src/lib/explanation/explanation.component.scss:3]: same as above

File: libs/event/src/lib/explanation/explanation.component.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { ChangeDetectionStrategy, Component, Inject, Input, Optional } from '@angular/core';
```
Comment (phpgeekfr) [inline libs/event/src/lib/explanation/explanation.component.ts:1]: path `libs/event/src/lib/explanation/explanation.component.ts` is not good. 
it should go into `libs/event/src/lib/components`

File: libs/event/src/lib/explanation/explanation.module.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+import { MatIconModule } from '@angular/material/icon';
+import { MatTabsModule } from '@angular/material/tabs';
+
+
+@NgModule({
+  imports: [
+    CommonModule,
+    MatButtonModule,
+    MatIconModule,
+    MatTabsModule,
```
Comment (phpgeekfr) [inline libs/event/src/lib/explanation/explanation.module.ts:17]: you dont need 
```
    MatIconModule,
    MatTabsModule,
```

also in `apps/catalog/catalog/src/app/marketplace/title/avails/explanation/explanation.module.ts` , `MatIconModule` is not needed

File: libs/event/src/lib/form/shell/shell.component.html [hunk 9]
```diff
@@ -32,13 +32,19 @@ <h2>Details</h2>
...
       <event-details-edit [form]="form" showPrivacy></event-details-edit>
     </mat-card>
-
-    <nav mat-tab-nav-bar>
-      <ng-container *ngFor="let tab of (tabs$ |async)">
-        <a mat-tab-link [routerLink]="tab.path" routerLinkActive #rla="routerLinkActive" [active]="rla.isActive">
-          {{ tab.label }}</a>
-      </ng-container>
-    </nav>
+    <div class="container-nav-button">
```
Comment (phpgeekfr) [inline libs/event/src/lib/form/shell/shell.component.html:35]:
```suggestion
    <div>
```

you don't need a class.

this should work:
```

form {
  div {
  display: flex;
  justify-content: space-between;
}
}
```

File: libs/event/src/lib/explanation/explanation.component.ts (hunk 7 above)
Comment (phpgeekfr) [inline libs/event/src/lib/explanation/explanation.component.ts:1]: `Input` import not used

File: libs/event/src/lib/explanation/explanation.component.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+  templateUrl: 'explanation.component.html',
+  styleUrls: ['./explanation.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ExplanationComponent {
+
+  constructor(
+    @Optional() private intercom: Intercom,
+    private dialogRef: MatDialogRef<ExplanationComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: { type: string },
```
Comment (phpgeekfr) [inline libs/event/src/lib/explanation/explanation.component.ts:16]:
```suggestion
    @Inject(MAT_DIALOG_DATA) public data: { type: EventTypes },
```

File: libs/event/src/lib/explanation/explanation.component.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
+import { ChangeDetectionStrategy, Component, Inject, Input, Optional } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { Intercom } from 'ng-intercom';
+
+@Component({
+  selector: 'catalog-movie-avails-explanation',
```
Comment (phpgeekfr) [inline libs/event/src/lib/explanation/explanation.component.ts:6]: rename the selector

File: libs/event/src/lib/form/shell/shell.component.scss [hunk 12]
```diff
@@ -35,4 +35,9 @@ form {
 
 mat-spinner {
   margin-top: 48px;
+}
+
+.container-nav-button {
+  display: flex;
+  place-content: space-between;
```
Comment (phpgeekfr) [inline libs/event/src/lib/form/shell/shell.component.scss:42]: use `justify-content: space-between;`

File: apps/catalog/catalog/src/app/marketplace/title/avails/explanation/explanation.component.html [hunk 13]
```diff
@@ -17,7 +17,7 @@ <h1>How It Works</h1>
       </mat-tab>
     </mat-tab-group>
 
-    <div mat-dialog-actions fxLayout="column" fxLayoutAlign="center center" fxLayoutGap="16px" class="actions">
+    <div mat-dialog-actions class="actions">
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/marketplace/title/avails/explanation/explanation.component.html:20]: this is the only `div` no need to use a class

File: apps/catalog/catalog/src/app/marketplace/title/avails/explanation/explanation.component.scss (hunk 5 above)
Comment (KvnMrs) [reply apps/catalog/catalog/src/app/marketplace/title/avails/explanation/explanation.component.scss:2]: I have the impression that the fxLayout was not really useful on this section
( in this video 'FlexLayoutModule' is well imported)

https://user-images.githubusercontent.com/86494393/175004829-cf6f23df-e0e3-483e-a46a-f004977e7d8c.mp4

Comment (phpgeekfr) [reply apps/catalog/catalog/src/app/marketplace/title/avails/explanation/explanation.component.scss:2]: ok maybe it is not needed but ` margin-right: 32px;` does not look the good css to use

File: libs/event/src/lib/components/explanation/explanation.component.html [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+  <h1>How It Works</h1>
+  <ng-container *ngIf="data.type === 'screening'; else meeting">
+    <p><strong>Screening</strong> Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc convallis augue vel ligula porttitor, sed ultrices tellus tincidunt. Donec nec lobortis nibh, et faucibus libero.</p>
+    <video controls src="assets/videos/screening_set_up.mp4"></video>    
+  </ng-container>
+  <ng-template #meeting>
+    <p><strong>Meeting</strong> Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc convallis augue vel ligula porttitor, sed ultrices tellus tincidunt. Donec nec lobortis nibh, et faucibus libero.</p>
+    <video controls src="assets/videos/meeting_set_up.mp4"></video>
+  </ng-template>
+  <div mat-dialog-actions class="actions">
```
Comment (phpgeekfr) [inline libs/event/src/lib/components/explanation/explanation.component.html:11]: remove class="actions"

File: libs/event/src/lib/components/explanation/explanation.component.html [hunk 15]
```diff
@@ -0,0 +1,17 @@
+<global-modal>
+  <h1>How It Works</h1>
+  <ng-container *ngIf="data.type === 'screening'; else meeting">
+    <p><strong>Screening</strong> Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc convallis augue vel ligula porttitor, sed ultrices tellus tincidunt. Donec nec lobortis nibh, et faucibus libero.</p>
+    <video controls src="assets/videos/screening_set_up.mp4"></video>    
+  </ng-container>
+  <ng-template #meeting>
+    <p><strong>Meeting</strong> Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc convallis augue vel ligula porttitor, sed ultrices tellus tincidunt. Donec nec lobortis nibh, et faucibus libero.</p>
+    <video controls src="assets/videos/meeting_set_up.mp4"></video>
+  </ng-template>
```
Comment (phpgeekfr) [inline libs/event/src/lib/components/explanation/explanation.component.html:10]: you can already prepare the component for slates

File: libs/event/src/lib/form/shell/shell.component.html [hunk 16]
```diff
@@ -32,13 +32,19 @@ <h2>Details</h2>
...
-    </nav>
+    <div>
+      <nav mat-tab-nav-bar>
+        <ng-container *ngFor="let tab of (tabs$ |async)">
+          <a mat-tab-link [routerLink]="tab.path" routerLinkActive #rla="routerLinkActive" [active]="rla.isActive">
+            {{ tab.label }}
+          </a>
+        </ng-container>
+      </nav>
+      <button *ngIf="form.value.type !== 'slate'" mat-button (click)="explain()">
```
Comment (phpgeekfr) [inline libs/event/src/lib/form/shell/shell.component.html:43]: you can already remove ngIf

**Review by phpgeekfr — APPROVED** [review]

**Pull request merged**